packages/template-retail-react-app/app/components/badge-group/index.jsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React from 'react'
+import PropTypes from 'prop-types'
+import {useStyleConfig, Box, Badge} from '@salesforce/retail-react-app/app/components/shared/ui'
+
+const BadgeGroup = (props) => {
+    const styles = useStyleConfig('BadgeGroup')
+
+    if (!props.badgeLabels) {
+        return
+    }
+    //TODO: check for duplicate labels
+    // If product is not provided/undefined then render the provided badge labels. This is useful if user already has the logic to get the badge labels outside the badge group
+    let filteredLabels = props.badgeLabels
+    if (props.product) {
+        // validate the provided badge labels against the product custom properties. This will allow users to use any boolean custom properties as badges
+        filteredLabels = props.badgeLabels.filter((item) => {
+            const propertyName = `c_is${item.label}`
+            return (
+                typeof props.product[propertyName] === 'boolean' &&
+                props.product[propertyName] === true
+            )
```

Comment:
I don't have strong feelings on this but just want to talk about the two limitations:

1. Customer must add a boolean custom property `c_is*` to products (this is buried in implementation detail, from developer's perspective, I'd imagine it's not easy to find out this. what can we do to better document "how to customize"?)
2. Only background color and label is customizable, other styles cannot be customized like font size, border radius, etc.

I'd imagine customers might do crazy things beyond the limitations. 

Some examples
1. What about dynamic labels? for example, product rankings. This might not be achievable from a boolean field, rather may depends on other product data
![CleanShot 2024-05-16 at 10 22 40@2x](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/10948652/e069bb2e-0755-408b-8b52-94bfcf415bab)
2. customized badge shape 

![CleanShot 2024-05-16 at 10 21 38@2x](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/10948652/e5a158a9-b300-40f3-a298-322db1046211)

Would we consider opening up the limitations:
1. instead of having an `c_is*` as input, can we use the entire product object?
2. instead of a fixed schema `PRODUCT_TILE_BADGE_LABELS` `{label: string, color: string}`, can we allow customers to pass in instances of React component, so customers have full control over what their Badge looks like.

Reply:
Current customization API:

```ts
type BadgeLabels = {
  label: string,
  color: string
}
```

Consider

```ts
type BadgeLabels = (product: Product) => JSX.Element[];
```

Reply:
i should have added this comment to the `ProductTile` component implementation, not in this component...

Reply:
1. We don't want to consider the entire product object props for few reasons
- product object has huge number of properties. It impacts performance 
- All the properties except custom properties have a predefined purpose. So it is unlikely to use any property other than custom properties (`c_`) for the badge. 
- We can consider all custom properties instead of just boolean but that makes the implementation complicated. So we decided to consider just the boolean custom properties.
2. The badge element is not passed to ProductTile to avoid too many props on the product tile. If users want to customized they can override the BadgeGroup. May be we can separate Badge into its own component?